test/spec/api/createVis/create-vis.spec.js

```diff
@@ -0,0 +1,344 @@
+var $ = require('jquery');
+var createVis = require('../../../../src/api/create-vis');
+var scenarios = require('./scenarios');
+var Loader = require('../../../../src/core/loader');
+
+describe('create-vis:', function () {
+  beforeEach(function () {
+    this.container = $('<div id="map">').css('height', '200px');
+    this.containerId = this.container[0].id;
+    $('body').append(this.container);
+  });
+
+  afterEach(function () {
+    this.container.remove();
+  });
+
+  it('should throw errors when required parameters are missing', function () {
+    expect(function () {
+      createVis();
+    }).toThrowError('a valid DOM element or selector must be provided');
+
+    expect(function () {
+      createVis('something');
+    }).toThrowError('a valid DOM element or selector must be provided');
+
+    expect(function () {
+      createVis(this.containerId);
+    }.bind(this)).toThrowError('a vizjson URL or object must be provided');
+
+    expect(function () {
+      createVis(this.container[0], 'vizjson');
+    }.bind(this)).not.toThrowError();
+
+    expect(function () {
+      createVis(this.containerId, 'vizjson');
+    }.bind(this)).not.toThrowError();
+  });
+
+  it('should use the given vis.json (instead downloading) when the visjson parameter is provided', function () {
+    spyOn(Loader, 'get');
+    var visJson = scenarios.load(0);
```

Comment:
This is pretty cool but we only have one scenario right now so we don't really need it... In any case, what about naming the scenario instead of using a number?